azurerm/internal/services/cosmos/tests/resource_arm_cosmosdb_sql_stored_procedure_test.go

```diff
@@ -0,0 +1,214 @@
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accountName := rs.Primary.Attributes["account_name"]
+		resourceGroupName := rs.Primary.Attributes["resource_group_name"]
+		databaseName := rs.Primary.Attributes["database_name"]
+		containerName := rs.Primary.Attributes["container_name"]
+
+		resp, err := client.GetSQLStoredProcedure(ctx, resourceGroupName, accountName, databaseName, containerName, name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on cosmosAccountsClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Cosmos Stored Procedure '%s' (account: '%s') does not exist", name, accountName)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMCosmosDbSqlStoredProcedure_base(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cosmos-%[2]d"
+  location = "%[1]s"
+}
+
+resource "azurerm_cosmosdb_account" "test" {
+  name                = "acctest-cosmos-%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  offer_type          = "Standard"
+  kind                = "%[3]s"
+
+  consistency_policy {
+    consistency_level = "%[4]s"
+  }
+
+  geo_location {
+    location          = azurerm_resource_group.test.location
+    failover_priority = 0
+  }
+}
+
+resource "azurerm_cosmosdb_sql_database" "test" {
+  name                = "acctest-sql-database-%[2]d"
+  resource_group_name = azurerm_cosmosdb_account.test.resource_group_name
+  account_name        = azurerm_cosmosdb_account.test.name
+}
+
+resource "azurerm_cosmosdb_sql_container" "test" {
+  name                = "acctest-sql-container-%[2]d"
+  resource_group_name = azurerm_cosmosdb_account.test.resource_group_name
+  account_name        = azurerm_cosmosdb_account.test.name
+  database_name       = azurerm_cosmosdb_sql_database.test.name
+}
+`, data.Locations.Primary, data.RandomInteger, string(documentdb.GlobalDocumentDB), string(documentdb.Session))
+}
+
+func testAccAzureRMCosmosDbSqlStoredProcedure_basic(data acceptance.TestData) string {
+	template := testAccAzureRMCosmosDbSqlStoredProcedure_base(data)
+
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_cosmosdb_sql_stored_procedure" "test" {
+  name                = "acctest-%[2]d"
+  resource_group_name = azurerm_cosmosdb_account.test.resource_group_name
+  account_name        = azurerm_cosmosdb_account.test.name
+  database_name       = azurerm_cosmosdb_sql_database.test.name
+  container_name      = azurerm_cosmosdb_sql_container.test.name
+
+  body = <<BODY
+  	function () {
+		var context = getContext();
+		var response = context.getResponse();
+		response.setBody('Hello, World');
+	}
+BODY
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMCosmosDbSqlStoredProcedure_update(data acceptance.TestData) string {
```

Comment:
This test was having issues running when updating it locally. Will continue troubleshooting the issues locally before pushing again. Wanted to get the other changes reviewed first